### DISCUSSION TITLE
How can I get logging in JSON format?

### DISCUSSION THREAD

**Discussion by justas200** [strimzi/strimzi-kafka-operator#10636] [Q&A] [answered]
I have been following some online tutorials. I've added 

1. a config map
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: json-logs-config
  namespace: {{ $root.Release.Namespace }} 
data:
  kafka-log4j.properties: |-
    # Set the root logger level to INFO and specify CONSOLE as the appender
    log4j.rootLogger=INFO, CONSOLE

    # Define the CONSOLE appender to use the JSONEventLayoutV1 layout
    log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
    log4j.appender.CONSOLE.layout=net.logstash.log4j.JSONEventLayoutV1

    # Kafka specific logging configurations
    log4j.logger.org.I0Itec.zkclient.ZkClient=INFO
    log4j.logger.org.apache.zookeeper=INFO
    log4j.logger.kafka=INFO
    log4j.logger.org.apache.kafka=INFO
    log4j.logger.kafka.request.logger=WARN, CONSOLE
    log4j.logger.kafka.network.Processor=OFF
    log4j.logger.kafka.server.KafkaApis=OFF
    log4j.logger.kafka.network.RequestChannel$=WARN
    log4j.logger.kafka.controller=TRACE
    log4j.logger.kafka.log.LogCleaner=INFO
    log4j.logger.state.change.logger=TRACE
    log4j.logger.kafka.authorizer.logger=INFO
```

and referenced it in Kafka resources
```
spec:
  kafka:
    logging:
    type: external
    valueFrom:
      configMapKeyRef:
        name: json-logs-config
        key: kafka-log4j.properties
```

Everything runs successfully but the logs are not in JSON format.
Follow up questions: how can I do this for KafkaConnect logs?

**Comment by katheris** [comment, marked as answer]
@justas200 I tried out the config map and configuration you shared and my Kafka pods are now logging in JSON format. Are you sure the operator is running? Did the Kafka pods roll after you made the change to the Kafka resource?

For Kafka Connect you can follow a similar process https://strimzi.io/docs/operators/latest/configuring#property-kafka-connect-logging-reference

**Reply by katheris** [reply]
It would also be useful if you could share what versions of the Strimzi operator and Kafka you are running

**Reply by justas200** [reply]
Thanks for the quick reply! I am running `quay.io/strimzi/operator:0.39.0`. 
Operator and the pods did all restart. 

I do see now that operator helm chart provides `logConfiguration` configuration (https://github.com/strimzi/strimzi-kafka-operator/tree/main/helm-charts/helm3/strimzi-kafka-operator#configuration) however trying to use JSONEventLayoutV1 there throws a startup error that the class is not found.

**Reply by justas200** [reply]
Sorry, it started working. Required a full cluster restart for the operator to catch up with changes. Thanks!
